modules/channel-web/src/views/lite/store/index.ts

```diff
@@ -385,13 +385,21 @@ class RootStore {
 
     this.api.updateAxiosConfig({ botId: this.config.botId, externalAuthToken: this.config.externalAuthToken })
     this.api.updateUserId(this.config.userId)
+    this.publishConfigChanged()
   }
 
   /** When this method is used, the user ID is changed in the configuration, then the socket is updated */
   @action.bound
   setUserId(userId: string): void {
     this.config.userId = userId
+    this.resetConversation()
     this.api.updateUserId(userId)
+    this.publishConfigChanged()
+  }
+
+  @action.bound
+  publishConfigChanged() {
+    this.postMessage('configChanged', JSON.stringify(this.config, undefined, 2))
```

Comment:
why stringify ?

Reply:
PostMessage seems to have trouble with complex objects, the config `{ overrides: { before_container: [] } }` ends up being an object `{ overrides: { before_container: {} } }` and it loses its array elements

Reply:
didn't know that. Thanks